code/go/0chain.net/chaincore/httpclientutil/http_client_util_test.go

```diff
@@ -0,0 +1,1050 @@
+		TransactionType   int
+		TransactionOutput string
+		OutputHash        string
+	}
+	type args struct {
+		handler Signer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+		want    *Transaction
+	}{
+		{
+			name:    "OK",
+			fields:  fields(*txn),
+			args:    args{handler: handler},
+			wantErr: false,
+			want:    &want,
+		},
+		{
+			name:   "ERR",
+			fields: fields(*txn),
+			args: args{
+				handler: func(h string) (string, error) {
+					return "", errors.New("")
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			txn := &Transaction{
+				Hash:              tt.fields.Hash,
+				Version:           tt.fields.Version,
+				ClientID:          tt.fields.ClientID,
+				PublicKey:         tt.fields.PublicKey,
+				ToClientID:        tt.fields.ToClientID,
+				ChainID:           tt.fields.ChainID,
+				TransactionData:   tt.fields.TransactionData,
+				Value:             tt.fields.Value,
+				Signature:         tt.fields.Signature,
+				CreationDate:      tt.fields.CreationDate,
+				Fee:               tt.fields.Fee,
+				TransactionType:   tt.fields.TransactionType,
+				TransactionOutput: tt.fields.TransactionOutput,
+				OutputHash:        tt.fields.OutputHash,
+			}
+			if err := txn.ComputeHashAndSign(tt.args.handler); (err != nil) != tt.wantErr {
+				t.Errorf("ComputeHashAndSign() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if !tt.wantErr {
+				assert.Equal(t, tt.want, txn)
+			}
+		})
+	}
+}
+
+func TestNewHTTPRequest(t *testing.T) {
+	t.Parallel()
+
+	var (
+		url  = "/"
+		data = []byte("data")
+		id   = "id"
+		pKey = "pkey"
+	)
+
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(data))
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	req.Header.Set("Access-Control-Allow-Origin", "*")
+	req.Header.Set("X-App-Client-ID", id)
+	req.Header.Set("X-App-Client-Key", pKey)
+
+	type args struct {
+		method string
+		url    string
+		data   []byte
+		ID     string
+		pkey   string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *http.Request
+		wantErr bool
+	}{
+		{
+			name: "OK",
+			args: args{
+				method: "",
+				url:    url,
+				data:   data,
+				ID:     id,
+				pkey:   pKey,
+			},
+			want:    req,
+			wantErr: false,
+		},
+		{
+			name: "ERR",
+			args: args{
+				url: string(rune(0x7f)),
+			},
+			want:    req,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			got, err := NewHTTPRequest(tt.args.method, tt.args.url, tt.args.data, tt.args.ID, tt.args.pkey)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewHTTPRequest() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != nil && !tt.wantErr {
+				assert.Equal(t, tt.want.URL, got.URL)
+				tt.want.URL = nil
```

Comment:
Race error here. tt.want is a *http.reqeuest object shared across all tests. So this is changing the tt.want for the remaing tests.

Either copy `*req` by value or move `req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(data))` inside the test.

Reply:
Pushed changes must solve this problem.